### PR TITLE
Improve timelock code

### PR DESCRIPTION
**Pull request by tcharding**
Done while working on a [timelock module](https://github.com/rust-bitcoin/rust-bitcoin/pull/994). This is all the initial patches (except one) from https://github.com/rust-bitcoin/rust-miniscript/pull/408 (which is a PR displaying usage of the new timelock module).

Note, does _not_ do the 'make `TimelockInfo` fields pub crate' change - I was unsure if this was correct.

### CODE REVIEW

**Review by apoelstra — DISMISSED** [review]
ACK e870519d0ff00cdfa6aef9fc36c543adcef57cbf

**Review by sanket1729 — APPROVED** [review]
ACK e870519d0ff00cdfa6aef9fc36c543adcef57cbf. Thanks for looking into this. Apart from the refactor here, there is also great value in someone looking at this code again.

File: src/miniscript/types/extra_props.rs [hunk 1]
```diff
@@ -89,7 +89,7 @@ impl TimeLockInfo {
             |mut timelock_info, sub_timelock| {
                 // If more than one branch may be taken, and some other branch has a requirement
                 // that conflicts with this one, set `contains_combination`
-                if k >= 2 {
+                if k > 1 {
```
[the remark this answers is not in the document]
Comment (tcharding) [reply src/miniscript/types/extra_props.rs:89]: Interesting, thanks for your thoughts. I tried to explain this if statement out loud to myself and I was unsure if I would say "if there are more than 1 ..." or "if there are 2 or more". I think I'd say "more than one" but I think that's just from reading `>` statements many times, so might be circular reasoning :) Amusingly I then opened the editor at this line and there is the following code comment
```
                // If more than one branch may be taken, and some other branch has a requirement
                // that conflicts with this one, set `contains_combination`.
```

**Comment by tcharding** [comment]
Force push is rebase only.

**Review by sanket1729 — APPROVED** [review]
ACK d1fdbaaf3f5af43f1d201a3d0dfbc15f877b0888. Reviwed the range-diff

**Pull request merged**